content/kb/using-streamlit/buttons.md

```diff
@@ -0,0 +1,412 @@
+---
+title: Button behavior and examples
+slug: /knowledge-base/using-streamlit/buttons
+---
+
+# Button behavior and examples
+
+## TLDR
+
+Buttons do not retain state. They return `True` on the page load resulting from
+their click and immediately return to `False` on the next load. If a displayed
+element is nested inside `if st.button('Click me'):`, the element will be
+visible when the button is clicked and disappear as soon as the user takes their
+next action. This is because the page reloads and the button becomes `False`.
+
+## Use cases for buttons
+
+When code is conditioned on a button's value, it will execute once in
+response to the button being clicked and not again (until the button is clicked
+again).
+
+### Nest inside of a button:
+
+- Transient messages that immediately disappear.
+- Once-per-click processes that saves data to session state, a file, or
+  a database.
+
+### Do not nest inside of a button:
+
+- Displayed items that should persist as the user continues.
+- Other widgets
+- Processes that neither modify session state nor write to a file/database.\*
+
+\* This can be appropriate when disposable results are desired. If you
+have a "Validate" button, that could be a process conditioned directly on a
+button. It could be used to create an alert to say 'Valid' or 'Invalid' with no
+need to keep that info.
+
+## Common logic with buttons
+
+### Show a quick message with a basic button
+
+If you want to give the user a quick button to check if an entry is valid,
+but not keep that check displayed as the user continues:
+
+```python
+import streamlit as st
+
+animal_shelter = ['cat', 'dog', 'rabbit', 'bird']
+
+animal = st.text_input('Type an animal')
+
+if st.button('Check availability') and animal.lower() in animal_shelter:
+    # This goes away if the user clicks the other button or changes the animal
+    st.write('We have that animal!')
+
+st.button('OK')
+```
```

Comment:
We should expand on what this code does and how it relates to the preceding sentence. "In the above/below example, ...."